modin/pandas/test/dataframe/test_map_metadata.py

```diff
@@ -746,30 +746,43 @@ def test_droplevel():
     [None, "col1", "name", ("col1", "col3"), ["col1", "col3", "col7"]],
     ids=["None", "string", "name", "tuple", "list"],
 )
-def test_drop_duplicates(data, keep, subset):
+@pytest.mark.parametrize("ignore_index", [True, False], ids=["True", "False"])
+def test_drop_duplicates(data, keep, subset, ignore_index):
```

Comment:
While the issue that this PR is fixing reveals a bug with dropping duplicates from a dataframe with a repeated index, none of the frames in `test_data_with_duplicates` has a repeated index. We should test that we are dropping the correct duplicates when the index has duplicates. For Modin right now we have:

```python
import modin.pandas as pd
import pandas

df = pd.DataFrame([[1], [2], [1]], index=[0, 0, 0])
# This gives an empty frame!
print(df.drop_duplicates())

pdf = pandas.DataFrame([[1], [2], [1]], index=[0, 0, 0])
# This drops just the second 1
print(pdf.drop_duplicates())
```

`test_data_with_duplicates_values` has just the data with no columns, so we can't add custom columns there. Could you please add another test case with a small reproducer as above for just this issue, and add a link to the github issue it's testing?